### PR TITLE
Loading using legacy h5 api now calls load_own_variables

### PR DESCRIPTION
**Pull request by Doch88**
https://github.com/keras-team/keras/issues/20147

```
So, I investigated it a bit and the problem is the keras.layers.Normalization layer, which is not finalized.
This is because keras.layers.Normalization has two weights loaded from the file, adapted_mean and adapted_std, but these are not used directly when making a prediction.
What is used directly are two variables, which are not added as weights, called mean and std, that are updated whenever finalize_state() is called.

And this is the problem. Theoretically, whenever loading weights from a file, finalize_state() should be called by load_own_variables().
But load_own_variables() is called only with the new saving API, when the weights are saved in the format ".weights.h5", while the efficientnet weights have the legacy extension ".h5".

Therefore, weights are loading, we have adapted_mean and adapted_std, but the actual mean and std are initialized with build(), i.e. a vector of zeros and a vector of ones.
```

### CODE REVIEW

**Comment by Doch88** [comment]
Closed the PR, since it would break a test.

**Comment by fchollet** [comment]
Hi @Doch88, thanks for the PR. Can you add instead a unit test that would currently fail but would start passing once the `Normalization` issue is resolved? I can use that test to make the fix.

**Comment by Doch88** [comment]
Hi @fchollet I made a new PR, including also the unit test:
https://github.com/keras-team/keras/pull/20168

Tests run well on my machine, let's see how it goes on the github pipeline
